usr/lib/ep11_stdll/ep11_specific.c

```diff
@@ -2917,6 +2922,175 @@ CK_RV ep11tok_aes_xts_pkey_check(STDLL_TokData_t *tokdata, CK_ATTRIBUTE_PTR attr
     return CKR_OK;
 }
 
+/* import a AES-XTS key, that is, make a blob for a AES XTS key
+ * that was not created by EP11 hardware, encrypt the key by the wrap key,
+ * unwrap it by the wrap key
+ */
+static CK_RV import_aes_xts_key(STDLL_TokData_t *tokdata, SESSION *sess,
+                                OBJECT *aes_xts_key_obj,
+                                CK_BYTE *blob, size_t *blob_size)
+{
+    ep11_private_data_t *ep11_data = tokdata->private_data;
+    CK_BYTE cipher[MAX_BLOBSIZE];
+    CK_ULONG clen = sizeof(cipher);
+    CK_BYTE csum[MAX_CSUMSIZE];
+    size_t cslen = sizeof(csum);
+    CK_BYTE iv[AES_BLOCK_SIZE];
+    size_t blob_size2 = *blob_size;
+    CK_MECHANISM mech = { CKM_AES_CBC_PAD, iv, AES_BLOCK_SIZE };
+    CK_RV rc;
+    CK_ATTRIBUTE_PTR p_attrs = NULL;
+    CK_ULONG attrs_len = 0;
+    CK_ATTRIBUTE_PTR new_p_attrs = NULL;
+    CK_ULONG new_attrs_len = 0;
+    CK_ATTRIBUTE *chk_attr = NULL, *attr = NULL;
+    unsigned char *ep11_pin_blob = NULL;
+    CK_ULONG ep11_pin_blob_len = 0;
+    ep11_session_t *ep11_session = (ep11_session_t*) sess->private_data;
+
```

Comment:
Should we maybe also check here if XTS is supported at all? I.e. if msa level is correct, and pkey is not disabled? Otherwise we import an XTS key but can't use it.

Maybe add
`if (ep11tok_is_mechanism_supported(tokdata, CKM_AES_XTS_KEY) != CKR_OK) ....`
here and return CKR_KEY_TYPE_INCONSISTENT if the XTS mech is not supported.